source/dol/bases/d_a_iceball.cpp

```diff
@@ -0,0 +1,506 @@
+#include <game/bases/d_a_iceball.hpp>
+#include <game/bases/d_a_player_manager.hpp>
+#include <game/bases/d_audio.hpp>
+#include <game/bases/d_actor.hpp>
+#include <game/bases/d_eff_actor_manager.hpp>
+#include <game/bases/d_effectmanager.hpp>
+#include <game/bases/d_s_stage.hpp>
+#include <game/bases/d_cd.hpp>
+#include <game/bases/d_bg.hpp>
+#include <game/framework/f_profile.hpp>
+#include <game/framework/f_profile_name.hpp>
+#include <game/sLib/s_GlobalData.hpp>
+#include <constants/sound_list.h>
+
+
+ACTOR_PROFILE(ICEBALL, daIceBall_c, 2);
+
+int daIceBall_c::sm_IceBallCount[4];
+int daIceBall_c::sm_IceBallAliveCount[4];
+
+template <>
+const daIceBall_c::GlobalData_t sGlobalData_c<daIceBall_c>::mData = {
+    0.75f,
+    1.05f,
+    0.6f,
+    120.0f
+};
+
+const sBcSensorPoint l_iceball_foot = { SENSOR_IS_POINT, 0, -0x3000 };
+const sBcSensorPoint l_iceball_head = { SENSOR_IS_POINT, 0, 0x3000 };
+const sBcSensorPoint l_iceball_wall = { SENSOR_IS_POINT, 0x3000, 0 };
+
+const sCcDatNewF l_fball_cc_data = {
+    0.0f, 0.0f,
+    6.0f, 6.0f,
+    CC_KIND_TAMA, CC_ATTACK_ICEBALL,
+    BIT_FLAG(CC_KIND_PLAYER_ATTACK) | BIT_FLAG(CC_KIND_ENEMY) |
+    BIT_FLAG(CC_KIND_BALLOON) | BIT_FLAG(CC_KIND_ITEM) | BIT_FLAG(CC_KIND_TAMA) | BIT_FLAG(CC_KIND_KILLER),
+    BIT_FLAG(CC_ATTACK_ICE_BREAK) | BIT_FLAG(CC_ATTACK_KOOPA_FIRE) | BIT_FLAG(CC_ATTACK_YOSHI_EAT),
+    CC_STATUS_NONE,
+    daIceBall_c::ccCallback_Iceball
+};
```

Comment:
```suggestion
const sCcDatNewF l_fball_cc_data = {
    0.0f, 0.0f,
    6.0f, 6.0f,
    CC_KIND_TAMA,
    CC_ATTACK_ICEBALL,
    BIT_FLAG(CC_KIND_PLAYER_ATTACK) | BIT_FLAG(CC_KIND_ENEMY) |
      BIT_FLAG(CC_KIND_BALLOON) | BIT_FLAG(CC_KIND_ITEM) | BIT_FLAG(CC_KIND_TAMA) | BIT_FLAG(CC_KIND_KILLER),
      BIT_FLAG(CC_ATTACK_ICE_BREAK) | BIT_FLAG(CC_ATTACK_KOOPA_FIRE) | BIT_FLAG(CC_ATTACK_YOSHI_EAT),
    CC_STATUS_NONE,
    daIceBall_c::ccCallback_Iceball
};
```
Trying to make clear when each field of the struct ends here, not sure if this helps in any way.